src/action/base/create_directory.rs

```diff
@@ -236,6 +247,36 @@ impl Action for CreateDirectory {
     }
 }
 
+// There are cleaner ways of doing this (eg `systemctl status $PATH`) however we need a widely supported way.
+async fn path_is_mountpoint(path: &Path) -> Result<bool, ActionErrorKind> {
+    let path_str = match path.to_str() {
+        Some(path_str) => path_str,
+        None => return Ok(false),
+    };
+
+    let output = execute_command(Command::new("mount").process_group(0)).await?;
+    let output_string = String::from_utf8(output.stdout).map_err(ActionErrorKind::FromUtf8)?;
+
+    // Each line looks like `portal on /run/user/1000/doc type fuse.portal (rw,nosuid,nodev,relatime,user_id=1000,group_id=100)`
+    for line in output_string.lines() {
+        let mut line_splitter = line.split(" on ");
+        match line_splitter.next() {
+            Some(_device) => (),
+            None => continue,
+        }
+        let destination_and_options = match line_splitter.next() {
+            Some(destination_and_options) => destination_and_options,
+            None => continue,
+        };
+        let trimmed = destination_and_options.trim();
+        if trimmed.starts_with(path_str) {
```

Comment:
I don't know if this is quite correct. If someone has a separate mountpoint for e.g. `/nix/store` but not `/nix`, wouldn't this improperly detect that? I think you'll want to split on space at that point to get the whole mountpoint.

Reply:
I think you're right!